Database.cpp

```diff
@@ -184,11 +215,17 @@ void Database::commit_task(uint64_t task_id) {
                 return; // suspicious, but maybe delayed task
             }
             ds->toggle_taint(change.parameter);
-            ds->on_disk_metadata_update();
+            ds->save();
+        } else if (change.type == DbChangeType::NewIterator) {
+            DatabaseName itname = DatabaseName::parse(db_base, change.obj_name);
+            load_iterator(itname);
+        } else if (change.type == DbChangeType::UpdateIterator) {
+            DatabaseName itname = DatabaseName::parse(db_base, change.obj_name);
+            uint64_t offset = std::atoi(change.parameter.c_str());
```

Comment:
:concerned:

Reply:
you think that was concerning? think again